jina/peapods/runtimes/jinad/__init__.py

```diff
@@ -53,92 +75,70 @@ async def _wait_for_cancel(self):
-        self.workspace_api.delete(id=self.args.workspace_id)
-        super().teardown()
-
-    def create_workspace(self):
-        """Create a workspace on remote (includes file upload & docker build)
-
-        :raises DaemonConnectivityError: if remote daemon is not reachable
-        :raises RuntimeError: if workspace creation fails
-        """
-        if not self.workspace_api.alive:
-            self.logger.error(
-                f'fail to connect to the daemon at {self.host}:{self.port_expose}, please check:\n'
-                f'- is there a typo in {self.host}?\n'
-                f'- on {self.host}, are you running `docker run --network host jinaai/jina:latest-daemon`?\n'
-                f'- on {self.host}, have you set the security policy to public for all required ports?\n'
-                f'- on local, are you behind VPN or proxy?'
-            )
-            raise DaemonConnectivityError
+        """Cancels the logstream task, removes the remote Pea & Workspace"""
+        self.logstream.cancel()
+        await self.client.peas.delete(id=self.pea_id)
+        # NOTE: don't fail if workspace deletion fails here
+        await self.client.workspaces.delete(id=self.workspace_id)
 
-        sleep = 2
-        retries = 100
-        for retry in range(retries):
-            workspace_status = self.workspace_api.get(id=self.args.workspace_id)
-            if not workspace_status:
-                raise DaemonConnectivityError
-            state = workspace_status.get('state', None)
-            if not state:
-                self.logger.info(
-                    f'creating workspace {colored(self.args.workspace_id, "cyan")} on remote. This might take some time.'
-                )
-                self.workspace_api.post(
-                    dependencies=self.args.upload_files,
-                    workspace_id=self.args.workspace_id,
-                )
-            elif state in [
-                RemoteWorkspaceState.PENDING,
-                RemoteWorkspaceState.CREATING,
-                RemoteWorkspaceState.UPDATING,
-            ]:
-                if retry % 10 == 0:
-                    self.logger.info(
-                        f'workspace {self.args.workspace_id} is getting created on remote. waiting..'
-                    )
-                time.sleep(sleep)
-            elif state == RemoteWorkspaceState.ACTIVE:
-                self.logger.success(
-                    f'successfully created a remote workspace: {colored(self.args.workspace_id, "cyan")}'
-                )
-                break
+    async def _sleep_forever(self):
+        """Sleep forever, no prince will come."""
+        await asyncio.sleep(1e10)
+
+    def _mask_args(self, args: 'argparse.Namespace'):
+        _args = copy.deepcopy(args)
+
+        # reset the runtime to ZEDRuntime or ContainerRuntime
+        if _args.runtime_cls == 'JinadRuntime':
```

Comment:
Can we somehow reuse the existing logic here for finding the correct runtime class ? This looks error prone to future changes in runtime class config?

Reply:
This is same as the previous logic, just moved from a different file.